packages/@aws-cdk/aws-ecs/README.md

```diff
@@ -1,2 +1,142 @@
-## The CDK Construct Library for AWS Elastic Container Service (ECS)
-This module is part of the [AWS Cloud Development Kit](https://github.com/awslabs/aws-cdk) project.
+## AWS Elastic Container Service (ECS) Construct Library
+
+This package contains constructs for working with **AWS Elastic Container
+Service** (ECS). The simplest example of using this library looks like this:
+
+```ts
+// Create an ECS cluster (backed by an AutoScaling group)
+const cluster = new ecs.Ec2Cluster(this, 'Cluster', {
+  vpc,
+  size: 3,
+  instanceType: new InstanceType("t2.xlarge")
+});
+
+// Instantiate ECS Service with an automatic load balancer
+const ecsService = new ecs.LoadBalancedEc2Service(this, 'Service', {
+  cluster,
+  memoryLimitMiB: 512,
+  image: ecs.DockerHub.image("amazon/amazon-ecs-sample"),
+});
+```
+
+### Fargate vs ECS
+
+There are two sets of constructs in this library; one to run tasks on ECS and
+one to run Tasks on fargate.
+
+- Use the `Ec2Cluster`, `Ec2TaskDefinition` and `Ec2Service` constructs to
+  run tasks on EC2 instances running in your account.
+- Use the `FargateCluster`, `FargateTaskDefinition` and `FargateService`
+  constructs to run tasks on instances that are managed for you by AWS.
+
+## Cluster
+
+An `Ec2Cluster` or `FargateCluster` defines the infrastructure to run your
+tasks on. If you create an ECS cluster, an AutoScalingGroup of EC2 instances
+running the latest ECS Optimized AMI will implicitly be created for you.
+
+You can run many tasks on a single cluster.
+
+To create a cluster, go:
+
+```ts
+const cluster = new ecs.FargateCluster(this, 'Cluster', {
+  vpc: vpc
+});
+```
+
+## TaskDefinition
+
+A `TaskDefinition` describes what a single copy of a **Task** should look like.
+A task definition has one or more containers; typically, it has one
+main container (the *default container* is the first one that's added
+to the task definition, and it will be marked *essential*) and optionally
+some supporting containers which are used to support the main container,
+doings things like upload logs or metrics to monitoring services.
+
+To add containers to a `TaskDefinition`, call `addContainer()`:
+
+```ts
+const taskDefinition = new ecs.FargateTaskDefinition(this, 'TaskDef', {
+  memoryMiB: '512'
+  cpu: 256
+});
+
+taskDefinition.addContainer('main', {
+  // Use an image from DockerHub
+  image: ecs.DockerHub.image('amazon/amazon-ecs-sample')
+});
+```
+
+### Images
+
+Images supply the software that runs inside the container. Images can be
+obtained from either DockerHub or from ECR repositories:
+
+* `ecs.DockerHub.image(imageName)`: use a publicly available image from
+  DockerHub.
+* `repository.getImage(tag)`: use the given ECR repository as the image
+  to start.
+
```

Comment:
Maybe, we can add some sugar to make this a bit more symmetric and discoverable:

```
ecs.Image.dockerHub(image)
esc.Image.ecr(repository, tag)
```

(I would still keep `repository.getImage(tag)`, but it would be nice if the usage from ECS's side would be symmetric)

Reply:
I would remove `repository.getImage(tag)` to avoid the dependency between ECR => ECS (that's the wrong direction)